### PR TITLE
[apptools-android] Support shared mode for apptools android

### PR DESCRIPTION
**Pull request by yunxliu**
Impacted tests(approved): new 3, update 66, delete 0
Unit test platform: [Android]
Unit test result summary: pass 66, fail 3, block 0

https://crosswalk-project.org/jira/browse/XWALK-4026

### CODE REVIEW

File: apptools/apptools-android-tests/README [hunk 1]
```diff
@@ -8,11 +8,13 @@ This test suite is for apptools-android-tests
 1. Connect Android devices to your localhost
 2. Need to edit the file "apptools-android-tests/arch.txt" according to the type of test device.
 If test device is "arm" platform, content of the file should be "arm", default value is "x86"
-3. Need to edit the file "apptools-android-tests/host.txt" according to the type of test host.
+3. Need to edit the file "apptools-android-tests/mode.txt" according to the type of build mode.
+If build "embedded" mode apk, content of the file should be "embedded", default value is "shared"
```
Comment (Honry) [inline apptools/apptools-android-tests/README]: No, I think default value is better to be "embedded"

File: apptools/apptools-android-tests/apptools/build.py [hunk 2]
```diff
@@ -58,7 +58,10 @@ def test_build_release(self):
 
     def test_build_missing_so_file(self):
         comm.setUp()
-        comm.create(self)
+        cmd = comm.HOST_PREFIX + comm.PackTools + \
+            "crosswalk-app create org.xwalk.test --android-crosswalk=" + \
```
Comment (Honry) [inline apptools/apptools-android-tests/apptools/build.py]: Where's the shared mode config?

File: apptools/apptools-android-tests/apptools/build.py [hunk 3]
```diff
@@ -92,7 +95,10 @@ def test_build_missing_so_file(self):
 
     def test_build_missing_both_so_file(self):
         comm.setUp()
-        comm.create(self)
+        cmd = comm.HOST_PREFIX + comm.PackTools + \
+            "crosswalk-app create org.xwalk.test --android-crosswalk=" + \
```
Comment (Honry) [inline apptools/apptools-android-tests/apptools/build.py]: ditto

File: apptools/apptools-android-tests/apptools/build.py (hunk 2 above)
Comment (yunxliu) [inline apptools/apptools-android-tests/apptools/build.py]: Due to the shared mode apk hasn't libxwalkcore.so file, so it isn't need to test.

**Comment by Honry** [comment]
LGTM

**Pull request merged**